tensorly/random/base.py

```diff
@@ -199,14 +199,10 @@ def random_tt(shape, rank, full=False, random_state=None, **context):
 
     # Initialization
     if rank[0] != 1:
-        message = "Provided rank[0] == {} but boundaring conditions dictatate rank[0] == rank[-1] == 1: setting rank[0] to 1.".format(
-            rank[0]
-        )
+        message = f"Provided {rank[0] = } but boundaring conditions dictatate rank[0] == rank[-1] == 1: setting rank[0] to 1."
         raise ValueError(message)
     if rank[-1] != 1:
-        message = "Provided rank[-1] == {} but boundaring conditions dictatate rank[0] == rank[-1] == 1: setting rank[-1] to 1.".format(
-            rank[0]
-        )
+        message = f"Provided {rank[-1] = } but boundaring conditions dictatate rank[0] == rank[-1] == 1: setting rank[-1] to 1."
```

Comment:
Changed to support Python 3.7
```suggestion
        message = f"Provided rank[-1] = {rank[-1]} but boundaring conditions dictatate rank[0] == rank[-1] == 1: setting rank[-1] to 1."
```